### DISCUSSION TITLE
Does not accept my PAT on a new system, but accepts the same PAT on an older one.

### DISCUSSION THREAD

**Discussion by camuso** [community/community#142169] [Code Security] [answered]
### Select Topic Area

Question

### Body

$ git push origin master
Username for 'https://github.com': camuso
Password for 'https://camuso@github.com': <pasted my PAT here>
remote: Invalid username or password. 
fatal: Authentication failed for 'https://github.com/camuso/scripts.git/'

**Comment by luannndev** [comment, marked as answer]
Hi camuso,

It looks like you're encountering an authentication issue with your Personal Access Token (PAT) on a new system. Here are a few things you can try to resolve the issue:

Token Scope: Make sure that your PAT has the correct scopes enabled for the actions you want to perform. For pushing to a repository, you typically need the repo scope.

Token Format: Double-check that you are using the PAT correctly as the password when prompted. It should be used instead of your GitHub password.

Credential Manager: If you are using a credential manager (like Git Credential Manager), it might be storing outdated credentials. You can try clearing the stored credentials and then re-entering your PAT.

Git Version: Ensure that you are using a compatible version of Git. Sometimes, updating Git can resolve authentication issues.

HTTPS vs. SSH: If you continue to have problems, consider switching to SSH for pushing to your repositories. You can generate an SSH key and add it to your GitHub account to authenticate without using a PAT.

Revoke and Regenerate PAT: If all else fails, try revoking your existing PAT and generating a new one, making sure to copy it correctly.

I hope one of these suggestions helps you resolve the authentication issue!

**Reply by camuso** [reply]
Scope:
I can push from the older system, so can't be the scope.

Format:
I have my token saved in a directory.
$ cat MyGitHubToken 
< copy to clipboard and paste to prompt >
No success.

Credential Manager:
Not using one.

git version 2.43.0
I can push changes from the older system

HTTP vs. SSH:
Changing to ssh fixed it!!!!!!!
MANY THANKS!!!
